go113_test.go

```diff
@@ -0,0 +1,140 @@
+// +build go1.13
+
+package errors
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestIs(t *testing.T) {
+	err := New("test")
+
+	type args struct {
+		err    error
+		target error
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "with stack",
+			args: args{
+				err:    WithStack(err),
+				target: err,
+			},
+			want: true,
+		},
+		{
+			name: "with message",
+			args: args{
+				err:    WithMessage(err, "test"),
+				target: err,
+			},
+			want: true,
+		},
+		{
+			name: "with message format",
+			args: args{
+				err:    WithMessagef(err, "%s", "test"),
+				target: err,
+			},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := Is(tt.args.err, tt.args.target); got != tt.want {
+				t.Errorf("Is() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+type customErr struct{}
+
+func (customErr) Error() string { return "" }
+
+func TestAs(t *testing.T) {
+	var err customErr
+
+	type args struct {
+		err    error
+		target interface{}
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "with stack",
+			args: args{
+				err:    WithStack(err),
+				target: new(customErr),
```

Comment:
So… there’s a lot of interesting features of `struct{}` types. In particular here, they can all point to the same address in memory. This means (per surprisingly), that all of these `target`s could be exactly the same value.

It would be better to mimic the [example from `errors.As`](https://golang.org/pkg/errors/#As) and use something like a `*os.PathError` or at least use something like a `type customError struct { msg string }` that will ensure that the type has a non-zero size. Then use a `*customError` which you then dereference for the `As(…)` call below.

You should then make sure that the `customError.msg` field is actually set to the expected value that you set inside your `baseError := &customError{ msg: "…" }` above.

Remember, the functionality of `As(…)` is not just that it returns true or false, but that it also assigns that matching value into the `target` parameter.

Reply:
I forgot this 

> Remember, the functionality of As(…) is not just that it returns true or false, but that it also assigns that matching value into the target parameter.

Let me add more check for this